File: showcase/discount-calculator.php
<?php

// Step 1: Ask the user for the amount of their purchase
$purchaseAmount = readline("Enter the amount of your purchase: ");

// Step 2: Check if the purchase amount is greater than $100
if ($purchaseAmount > 100) {
    // Step 3: Apply a 10% discount to the purchase amount
    $discountedAmount = $purchaseAmount * 0.9;
} elseif ($purchaseAmount > 50) {
    // Step 4: Check if the purchase amount is greater than $50
    // Step 5: Apply a 5% discount to the purchase amount
    $discountedAmount = $purchaseAmount * 0.95;
} else {
    $discountedAmount = $purchaseAmount;
}

// Step 6: Print the discounted amount
echo "Discounted amount: $" . $discountedAmount . "\n";

File: showcase/discount-calculator.c
#include <stdio.h>

int main()
{
  float purchaseAmount, discountAmount, discountedPrice;

  // Prompt the user to enter the amount of their purchase
  printf("Enter the amount of your purchase: ");
  scanf("%f", &purchaseAmount);

  // Check if the purchase amount is greater than $100
  if (purchaseAmount > 100)
  {
    // Calculate the discount
    discountAmount = purchaseAmount * 0.1;
  }
  else if (purchaseAmount > 50)
  {
    // Calculate the discount
    discountAmount = purchaseAmount * 0.05;
  }
  else
  {
    // No discount
    discountAmount = 0;
  }

  // Calculate the discounted price
  discountedPrice = purchaseAmount - discountAmount;

  // Print the discounted price
  printf("Discounted price: $%.2f\n", discountedPrice);

  return 0;
}